### PR TITLE
Fix "video not available" being shown to the user when when YouTube starts asking for captcha resolution due to receiving too many requests from the same IP.

### PR DESCRIPTION
**Pull request by kovan**
Fix "video not available" being shown to the user when when YouTube starts asking for captcha resolution due to receiving too many requests from the same IP. Show instead an appropiate message.

To be able to keep making requests, the captcha must be solved in a browser and the browser cookie must be passed to youtube-transcript-api.

See issue #60

### CODE REVIEW

**Comment by jdepoix** [comment]
Hi @kovan, great contribution, thank you very much!

Actually I wasn't even aware that it is possible to continue doing requests by solving a captcha. Do you know more about how this works? Is that tied to a session or the IP? If it is tied to the IP it would should be enough to solve it in the browser without using a cookie, right? Can you share some insights on how this works?

**Comment by kovan** [comment]
Good questions. 
It is by IP, they say it in the page that they show when it happens and also I have verified it myself. They say this:

```
Sorry for the interruption. We have been receiving a large volume of requests from your network.

To continue with your YouTube experience, please fill out the form below.
```

But when you solve the captcha, they don't unban the IP, they only store a cookie in the browser that allows you to visit Youtube while you have it. 
For the IP to be unbanned, my experience is that it has to pass some time, not much, only some minutes.

**Comment by jdepoix** [comment]
Thanks you for the information.
If you say that it's only a couple of minutes we could implement some retry mechanism when this error is catched and keep retrying with 5 minute timeout or something like that. That probably is more reliant (and even faster) than solving the captcha in the browser and restarting the script with the cookie. Do you have a more concrete idea on how long it usually takes and what a good timeout time would be? Also, does the ban period seem to be consistent in length?

Also, I think it would be nice to add more information to the error message like that 
- the user can use the cookie from the browser after solving the captcha
- the user can just wait a few minutes before continuing to hammer YouTube 😄 

And yes, I also doubt that we can generate the cookie in any way. For sure, that's an ID and the information that the captcha has been solved is on their servers. Doesn't hurt having a look tho, maybe it's easier than we think...

**Comment by kovan** [comment]
Forget about waiting a few minutes and you are unbanned, probably I wasn't testing properly (maybe I was using the cookie to make the request without noticing), because I have been testing more and with different intervals between requests, and my IP still banned since yesterday :/. So YouTube wins. 
I have improved the message a bit. IMHO there is not much more to do here. It is still a good improvement because now the user knows what is happening and what his options are.

File: youtube_transcript_api/_errors.py [hunk 1]
```diff
@@ -37,7 +37,13 @@ def cause(self):
...
 class VideoUnavailable(CouldNotRetrieveTranscript):
     CAUSE_MESSAGE = 'The video is no longer available'
-
+    
+class TooManyRequests(CouldNotRetrieveTranscript):
+    CAUSE_MESSAGE = ('YouTube is receiving too many requests from this IP, '
+        'and now requires that a captcha must be solved in order to continue. '
+        'You can solve the captcha in a browser and pass the generated cookie file to youtube-transcript-api, '
+        'or you can use a different IP, or maybe wait for the ban to be lifted.'
+        )
```
Comment (jdepoix) [inline youtube_transcript_api/_errors.py:46]: Just a tiny suggestion on the wording/formatting to make the options more clear:
```
YouTube is receiving too many requests from this IP and now requires solving a captcha to continue. One of the following things can be done to work around this:\n
 - Manually solve the captcha in a browser and export the cookie. Read here how to use that cookie with youtube-transcript-api: https://github.com/jdepoix/youtube-transcript-api#cookies\n
 - Use a different IP address\n
 - Wait until the ban on your IP has been lifted
```

**Comment by jdepoix** [comment]
Yeah, this absolutely is a great improvement! I wanted to implement this for a while since a lot of users open issues, as the error messages are not descriptive in this case. So this is an awesome contribution! 😊

**Comment by jdepoix** [comment]
@kovan I will merge this now, however, I probably won't get around to creating a new release until next week as I am quite busy atm. Once again, thank you very much for the contribution! 😊

**Pull request merged**